q2_types/_feature_data.py

```diff
@@ -0,0 +1,183 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+
+import skbio
+import skbio.io
+import pandas as pd
+
+import qiime
+from qiime.plugin import SemanticType, DataLayout, FileFormat
+from .plugin_setup import plugin
+
+
+FeatureData = SemanticType('FeatureData', field_names='type')
+
+Taxonomy = SemanticType('Taxonomy', variant_of=FeatureData.field['type'])
+
+Sequence = SemanticType('Sequence', variant_of=FeatureData.field['type'])
+
+AlignedSequence = SemanticType('AlignedSequence',
+                               variant_of=FeatureData.field['type'])
+
+
+class TaxonomyFormat(FileFormat):
+    name = 'taxonomy'
+
+    @classmethod
+    def sniff(cls, filepath):
+        with open(filepath, 'r') as fh:
+            count = 0
+            while count < 10:
+                line = fh.readline()
+                if line == '':
+                    break
+                elif line.startswith('#'):
+                    continue
+                else:
+                    cells = line.split('\t')
+                    if len(cells) < 2:
+                        return False
+                    count += 1
+            return False if count == 0 else True
```

Comment:
Oooh, I see... so that already is some non-obvious behavior! Add a comment indicating that goal and this is fine. (I would still prefer to see the return not be a ternary though as sniffers usually fit the inevitable conclusion pattern so well)